src/ray/util/tests/container_util_test.cc

```diff
@@ -14,18 +14,69 @@
 
 #include "ray/util/container_util.h"
 
+#include <sstream>
+#include <tuple>
+
 #include "gtest/gtest.h"
 
 namespace ray {
 
+template <typename T>
+std::string debug_string_to_string(const T &t) {
+  std::ostringstream ss;
```

Comment:
I think this is inefficient, your implementation would call https://en.cppreference.com/w/cpp/io/basic_stringstream/str (1).

Why not 
```
return std::move(ss).str();
```

Reply:
Prefer `absl::StreamFormat`, which is optimized for streaming usage

Doc:
```
absl::StreamFormat() returns an object that can be efficiently streamed to a std::ostream, such as I/O or files.
```

https://abseil.io/docs/cpp/guides/format

Reply:
used std::move.